src/app.js

```diff
@@ -1,13 +1,277 @@
+//   $('.error-messages').empty();
+//
+//   tripList.add(trip);
+//   $.modal.close();
+// };
+
+const addTrip = function addTrip(event) {
+  event.preventDefault();
+  const trip = new Trip(readTripForm());
+
+  // client side validations
+  if (trip.isValid()) {
+    trip.save({}, {
+      success: () => {
+        if ($('#add-trip-form p').length) {
+          $('#add-trip-form p').remove();
+        }
+
+        allTrips.add(trip);
+        $.modal.close();
+
+        renderSuccess('Trip successfully added', $('main > .status-messages'));
+      },
+      // success: successfulSave,
+      error: (model, response) => {
+        handleValidationErrors(response.responseJSON.errors, $('#add-trip-form'));
+      }
+      // error: failedSave
+    });
+  } else {
+    handleValidationErrors(trip.validationError, $('#add-trip-form'));
+  }
+};
+
+const loadResForm = function loadTripForm(tripId) {
+  const html = `
+  <form id="add-res-form" data-id="${tripId}" action="https://ada-backtrek-api.herokuapp.com/trips/${tripId}/reservations" method="post">
+    <section class="status-messages">
+    </section>
+    <div class="name grid-x">
+      <label class="small-6 cell">Name</label>
+      <input type="text" id="res-name" name="name" />
+    </div>
+
+    <div class="age grid-x">
+      <label class="small-6 cell">Age</label>
+      <input type="number" id="age" name="age" />
+    </div>
+
+    <div class="email grid-x">
+      <label class="small-6 cell">Email</label>
+      <input type="email" id="email" name="email" />
+    </div>
+
+    <button class="button confirm" type="submit" id="add-reservation">Reserve this Trip</button>
+    <button class="button cancel" type="reset" id="cancel-reservation">Cancel</button>
+  </form>`;
+
+  $('.reservation').append(html);
+};
+
+const addRes = function addRes(event) {
+  event.preventDefault();
+
+  const res = new Reservation(readResForm());
+
+  // client-side validations
+  if (res.isValid()) {
+    res.save({}, {
+      success: () => {
+        // if ($('#add-res-form p').length) {
+          $('#add-res-form p').remove();
+        // }
+        renderSuccess('Spot reserved', $('#add-res-form .status-messages'));
+        $('#add-res-form').trigger('reset');
+      },
+      error: (model, response) => {
+        handleValidationErrors(response.responseJSON.errors, $('#add-res-form'));
+      }
+    });
+  }
+  else {
+    handleValidationErrors(res.validationError, $('#add-res-form'));
+  }
+};
+
+const sortTrips = function sortTrips() {
+  const fields = ['id', 'name', 'continent', 'category', 'cost', 'weeks'];
+
+  $('.current-sort-field').removeClass('current-sort-field');
+  $(this).addClass('current-sort-field');
+
+  const classes = $(this).attr('class').split(/\s+/);
+
+  classes.forEach((className) => {
+    // if already sorted, sort desc
+    if (fields.includes(className)) {
+      if (className === allTrips.comparator) {
+        allTrips.models.reverse();
```

Comment:
This is a temporary change -- Backbone won't "know" that we've reversed the sort order for this comparator.

As a result, when Backbone decides to re-sort the collection for us (such as when a new trip is added), it will go back to the non-reversed order. How could we implement the sorting for our Backbone collection such that it will keep track of whether the sort was reversed or not?